CorsixTH/Src/sdl_core.cpp

```diff
@@ -296,6 +297,10 @@ int l_mainloop(lua_State* L) {
     lua_gc(L, LUA_GCSTEP, 2);
   }
 
+  if (wait_error != 0) {
+    std::fprintf(stderr, "%s\n", SDL_GetError());
+  }
```

Comment:
A few lines above do more `PollEvent` (274, 293), does the error survive that?

Reply:
High level structure is not very visible imho, but otherwise LGTM

Reply:
Yes, the complexity of this function is too high. Put it in the refactor list :+1: